### ISSUE TITLE
Use the SAP Cloud SDK in Azure AppService

### ISSUE THREAD

**Issue by MartinPankraz** [SAP/cloud-sdk#1066]
May we use the use the SAP Cloud SDK in Azure AppService? The public documentation doesn't mention execution environments outside of BTP. Could you outline which parts are usable and which parts are not?

**Comment by FrankEssenberger** [comment]
Hi @MartinPankraz, 

we have not tried this so I will give an answer just from my understanding not E2E tested. But since we support k8s and also have some test for frontend support we have a rough overview what should work and what not. 

- Destination retrieval either explicit or via the `execute()` method should work if the environment variables for accessing the destination service are present at runtime. But if you are not on BTP, I guess you do not want to use the destination service of BTP. Here some [destination docs](https://sap.github.io/cloud-sdk/docs/js/features/connectivity/destination) 
- The generated [Odata](https://sap.github.io/cloud-sdk/docs/js/features/odata/overview) or [OpenApi](https://sap.github.io/cloud-sdk/docs/js/features/openapi/overview) clients should work but as mentioned above you have to provide the destination object either in the execute call or [registerDestination()](https://sap.github.io/cloud-sdk/docs/js/features/connectivity/destination-cache-isolation#register-destination-cache) 

So to summarize you can generate typed clients with the SDK but the connectivity is not so smooth when you do not have the destination serivce. However, if you build the destination object on your own you can use the generated clients to retrieve data from SAP systems as with any other client.

I hope this answers your question?

Best
Frank

**Comment by MartinPankraz** [comment]
@FrankEssenberger thanks for the swift and detailed response! How about licensing and bundling the SDK with other code? I see Apache 2.0, but what about the SAP proprietary parts? Besides technically being possible are customers allowed to run the SDK outside of BTP?

**Comment by FrankEssenberger** [comment]
I am no legal person to be honest. Since we are open source I think you are allowed to use it. @marikaner do you know more details on this?

**Comment by marikaner** [comment]
Hey @MartinPankraz the SAP proprietary parts of the SDK are under the [SAP Developer license agreement](https://tools.eu1.hana.ondemand.com/developer-license-3_1.txt). This allows you use to build applications for customers. It does not restrict you to use SAP BTP. There are however some other restrictions, that mainly aim to protect the Intellectual Property of SAP (see 1. LICENSE in the agreement).

That said, I want to stress, that I am not a representative of SAP's legal department either, therefore I cannot guarantee that my understanding is 100% correct.

I hope that answers the question?
